File: src/main/java/fr/spoutnik87/musicbot_rest/repository/ServerRepository.java
package fr.spoutnik87.musicbot_rest.repository;

import fr.spoutnik87.musicbot_rest.model.Bot;
import fr.spoutnik87.musicbot_rest.model.Server;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ServerRepository extends JpaRepository<Server, Long> {

  Optional<Server> findByBot(Bot bot);

  Optional<Server> findByUuid(String uuid);
}

File: src/main/kotlin/fr/spoutnik87/musicbot_rest/repository/ServerRepository.kt
package fr.spoutnik87.musicbot_rest.repository

import fr.spoutnik87.musicbot_rest.model.Server
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import org.springframework.transaction.annotation.Transactional

@Repository
interface ServerRepository : JpaRepository<Server, Long> {
    @Transactional(readOnly = true)
    fun findByUuid(uuid: String): Server?
    @Transactional(readOnly = true)
    fun findByGuildId(guildId: String): Server?
}